aten/src/ATen/native/cuda/UnaryGeometricKernels.cu

```diff
@@ -150,15 +152,22 @@ void cosh_kernel_cuda(TensorIterator& iter) {
   });
 }
 
+// We manually override tanh because std::tanh does not work with thrust::complex types.
 template<typename scalar_t>
 __host__ __device__ static inline scalar_t tanh_wrapper(scalar_t v) {
   return ::tanh(v);
 }
 
+template<typename T>
+__host__ __device__ static inline thrust::complex<T> tanh_wrapper(thrust::complex<T> v) {
```

Comment:
`__host__ __device__ static inline c10::complex<T> tanh_wrapper(c10::complex<T> v)`

Reply:
Done!

Reply:
Reverted back to `thrust::complex<T>` till the time full migration from `std::complex` to `c10::complex` is complete. :)